### DISCUSSION TITLE
Comment Colors!

### DISCUSSION THREAD

**Discussion by utkonos** [Vector35/binaryninja-api#5243] [Ideas] [answered]
As I go through my RE process, I make comments about what I think the malware sample is doing. The first pass is typically heavy with hypotheses rather than definitive descriptions of what the malware capabilities and behavior are. As my process advances, I return to older comments and edit, tighten, adjust, and finalize them.

Today, all comments are the same color.

It would be helpful to be able to change the color of a particular comment. I would be able to have a "draft/hypothetical" comment color which I would have as the default color. And then I would be able to change the color of comments that I have finalized. Having multiple colors would make differentiating among comment types much easier.

Thougts on implementation:

1. There should be a configuration setting for the default color. The default would be today's comment color so as to not interrupt any other users' processes and expectations.
2. One wants comments to be visually unobtrusive like comments are now. Therefore the color palate for comments should only be slightly different than the color they are now. It should be just enough for visual differentiation but not so much that they distract from the disassembly or IL etc.
3. In the Enter Comment popup, at the bottom, there should be a color selector. The selector would be set to the configuration's default set in Preferences. One can change the selector to a different color for the comment being edited.
4. There should be API extensions (optional parameters to current functions) for creating and editing a comment via API. This would allow me to make right click plugin menu items like "Finalize Comment" that fit my process, allowing me to change the color of a comment however I need.

Thoughts?

**Comment by plafosse** [comment, marked as answer]
This would be a good feature. Adding this as an issue
